### PR TITLE
fix(payment): PI-1647 [Affirm] When returning to a cart with digital …

### PR DESCRIPTION
**Pull request by bc-sebastianszafraniec**
…item, Affirm is blocked with message "You provided an invalid shipping name."

## What?
Fix of the issue with Affirm transactions with digital item after moving to checkout and back to cart

## Testing / Proof
Tested manually 

https://github.com/user-attachments/assets/61ddcdd1-8ac3-4993-bce7-6c940bd622c6

@bigcommerce/team-checkout @bigcommerce/team-payments

### CODE REVIEW

File: packages/affirm-integration/src/affirm-payment-strategy.ts [hunk 1]
```diff
@@ -129,13 +131,19 @@ export default class AffirmPaymentStrategy implements PaymentStrategy {
         const amountTransformer = new AmountTransformer(order.currency.decimalPlaces);
         const billingAddress = this.getBillingAddress();
 
+        const retrievedShippingAddress = this.getShippingAddress();
+        const shippingAddress =
+            itemsRequireShipping(cart, config) && retrievedShippingAddress
```
Comment (vitalii-koshovyi) [inline packages/affirm-integration/src/affirm-payment-strategy.ts:136]: please cover it by test

Comment (bc-sebastianszafraniec) [reply packages/affirm-integration/src/affirm-payment-strategy.ts:136]: Sure, I will add it soon

Comment (bc-sebastianszafraniec) [reply packages/affirm-integration/src/affirm-payment-strategy.ts:136]: I added the test

**Comment by vitalii-koshovyi** [comment]
Thanks for adding test! Ping me after https://github.com/bigcommerce/checkout-sdk-js/pull/2687 merge, I'll approve this PR

**Review by vitalii-koshovyi — APPROVED** [review]

**Pull request merged**